plugin/unmarshal/unmarshal.go

```diff
@@ -675,26 +677,60 @@ func (p *unmarshal) field(file *descriptor.FileDescriptorProto, msg *generator.D
 			p.P(`}`)
 			p.P(`m.`, fieldname, ` = &`, p.OneOfTypeName(msg, field), `{v}`)
 		} else if generator.IsMap(file, field) {
-			mapMsg := generator.GetMap(file, field)
-			keyField, valueField := mapMsg.GetMapFields()
+			m := p.GoMapType(nil, field)
+			mapgoTyp, keyField, valueField, keyAliasField, valueAliasField := m.GoType, m.KeyField, m.ValueField, m.KeyAliasField, m.ValueAliasField
```

Comment:
This is a bit nit picky, but it seems like it is possible to just use m.KeyField, etc rather than assign them to other variables that basically have the same name.  What do you think?